src/components/PostDetails.tsx

```diff
@@ -1,117 +1,138 @@
+  // changeUser: boolean,
+};
+
+const { DeleteComment, LoadingComments, AddNewComment } = ErrorMessage;
+
+export const PostDetails: React.FC<Props> = ({
+  comments,
+  isProcessing,
+  setComments,
+  activePost,
+  setIsProcessing,
+  setErrorMessage,
+  errorMessage,
+  // changeUser,
+}) => {
+  const [writeCommentButton, setWriteCommentButton] = useState(false);
+  const [showError, setShowError] = useState(false);
+
+  const onRemoveComment = (id: number) => {
+    client.delete(`/comments/${id}`)
+      .catch(() => setErrorMessage(ErrorMessage.DeleteComment));
+    setComments(comments.filter(item => item.id !== id));
+  };
+
+  useEffect(() => setWriteCommentButton(false), [activePost]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setShowError([DeleteComment, AddNewComment, LoadingComments]
+        .includes(errorMessage));
+    }
+  }, [errorMessage]);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${activePost?.id}: ${activePost?.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {activePost?.body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
+          {isProcessing === ShowLoader.SideBar && <Loader />}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
+          { showError && errorMessage
+          && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              {errorMessage}
             </div>
+          )}
+          {isProcessing !== ShowLoader.SideBar && (
+            <>
+              {comments.length === 0
+                ? (
+                  <p className="title is-4" data-cy="NoCommentsMessage">
+                    No comments yet
+                  </p>
+                )
+                : (<p className="title is-4">Comments:</p>)}
```

Comment:
```suggestion
              {comments.length ? (
                  <p className="title is-4" data-cy="NoCommentsMessage">
                    No comments yet
                  </p>
                ) : (
                  <p className="title is-4">Comments:</p>
                )}
```

Reply:
with this approach, I have an extra number 0 in the application.